src/server/handlers/launchpad.js

```diff
@@ -97,11 +105,24 @@ export const setMemcached = (value) => {
   memcached = value;
 };
 
+class PreparedError extends Error {
+  constructor(status, body) {
+    super();
+    this.status = status;
+    this.body = body;
+  }
+}
+
 const responseError = (res, error) => {
-  if (error.response) {
+  if (error.status && error.body) {
+    // The error comes with a prepared representation.
+    return res.status(error.status).send(error.body);
+  } else if (error.response) {
     // if it's ResourceError from LP client at least for the moment
     // we just wrap the error we get from LP
-    return error.response.text().then(text => {
+    // XXX cjwatson 2016-12-22: Perhaps refactor to use the PreparedError
```

Comment:
I agree that it's a bit confusing right now with all the 'layers' of error we have (ResourceError, PreparedError, vanilla Errors...).

As ResourceError already seems to have a reference to response, maybe it could already have status and body set (be 'PreparedError' or at least in compatible format).